src/App.tsx

```diff
@@ -1,14 +1,57 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const defaultTodo = {
+    id: 0,
+    title: '',
+    completed: false,
+    userId: 0,
+  };
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState(defaultTodo);
+  const [query, setQuery] = useState('');
+  const [sortType, setSortType] = useState('all');
+
+  const normalizedQuery = query.toLowerCase();
+
+  useEffect(() => {
+    getTodos().then(todosFromServer => setTodos(todosFromServer));
+  }, []);
+
+  const getFilteredTodos = () => {
+    let filteredTodos = [...todos];
+
+    switch (sortType) {
+      case 'active':
```

Comment:
let`s make enum type for sortType and use it as a map
```suggestion
      case SortType.ACTIVE:
```